src/main/kotlin/cacophony/grammars/Analysis.kt

```diff
@@ -1,44 +1,169 @@
 package cacophony.grammars
 
 import cacophony.automata.DFA
+import kotlin.math.min
 
-typealias DFAStateReference<StateType, AtomType, ResultType> = Pair<StateType, DFA<StateType, AtomType, ResultType>>
+typealias DFAStateReference<StateType, SymbolType, ResultType> =
+    Pair<StateType, DFA<StateType, SymbolType, ResultType>>
 
-fun <StateType, SymbolType> findNullable(
-    automata: Map<SymbolType, DFA<StateType, SymbolType, Production<SymbolType>>>,
-): Collection<DFAStateReference<StateType, SymbolType, Production<SymbolType>>> {
-    TODO("Not implemented")
+sealed class Entity<out Q, out A, out R> {
```

Comment:
`Entity` is such a generic name, please change it to something reflecting what is really is for

Reply:
Changed to GeneralizedSymbol